RLTSquare/CustomEmailSender/Controller/Router.php

```diff
@@ -0,0 +1,196 @@
+ */
+class Router implements RouterInterface
+{
+    /**
+     *
+     */
+    const Custom_Email_Config_Path = 'custom_email/general/enable';
+    /**
+     *
+     */
+    const From_Email_Config_Path = 'custom_email/general/from_email';
+    /**
+     *
+     */
+    const From_Name_Config_Path = 'custom_email/general/from_name';
+    /**
+     *
+     */
+    const To_Email_Config_Path = 'custom_email/general/to_email';
+    /**
+     * @var PageFactory
+     */
+    protected PageFactory $pageFactory;
+    /**
+     * @var Logger
+     */
+    protected Logger $logger;
+    /**
+     * @var TransportBuilder
+     */
+    protected TransportBuilder $transportBuilder;
+    /**
+     * @var StoreManagerInterface
+     */
+    protected StoreManagerInterface $storeManager;
+    /**
+     * @var StateInterface
+     */
+    protected StateInterface $inlineTranslation;
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected ScopeConfigInterface $scopeConfig;
+    /**
+     * @var BackendTemplate
+     */
+    protected BackendTemplate $emailTemplate;
+    /**
+     * @var ActionFactory
+     */
+    private ActionFactory $actionFactory;
+
+    /**
+     * Router constructor.
+     *
+     * @param ActionFactory $actionFactory
+     * @param PageFactory $pageFactory
+     * @param Logger $logger
+     * @param TransportBuilder $transportBuilder
+     * @param StoreManagerInterface $storeManager
+     * @param StateInterface $state
+     * @param ScopeConfigInterface $scopeConfig
+     * @param BackendTemplate $emailTemplate
+     */
+    public function __construct(
+        ActionFactory         $actionFactory,
+        PageFactory           $pageFactory,
+        Logger                $logger,
+        TransportBuilder      $transportBuilder,
+        StoreManagerInterface $storeManager,
+        StateInterface        $state,
+        ScopeConfigInterface  $scopeConfig,
+        BackendTemplate       $emailTemplate
+    ) {
+        $this->actionFactory = $actionFactory;
+        $this->pageFactory = $pageFactory;
+        $this->logger = $logger;
+        $this->transportBuilder = $transportBuilder;
+        $this->storeManager = $storeManager;
+        $this->inlineTranslation = $state;
+        $this->scopeConfig = $scopeConfig;
+        $this->emailTemplate = $emailTemplate;
+    }
+
+    /**
+     * @param RequestInterface $request
+     * @return ActionInterface|null
+     */
+    public function match(RequestInterface $request): ?ActionInterface
+    {
+        $identifier = trim($request->getPathInfo(), '/');
+
+        if (strpos($identifier, 'rltsquare') !== false) {
+            $request->setModuleName('custom_email');
+            $request->setControllerName('index');
+            $request->setActionName('index');
+            $request->setParams([
+                'first_param' => 'first_value',
+                'second_param' => 'second_value'
+            ]);
+
+            $isEnabled = $this->scopeConfig->getValue(
```

Comment:
Unnecessary/inappropriate code - this "is enabled"-check code would probably make more sense if it wrapped around matching algorithm as well, not just email sending. But overall, seems unnecessary.